service.go

```diff
@@ -423,16 +442,62 @@ func populateService(k8sService *k8score.Service, acks map[string]string) servic
 			log.WithError(err).Warnf("Cannot parse isDaemon label value for service with name %s.", serviceName)
 		}
 	}
+	appPort := getAppPortForService(k8sService)
+	hs.populateSystemCodeForService(serviceName, appPort)
 
 	return service{
 		name:        serviceName,
-		appPort:     getAppPortForService(k8sService),
+		appPort:     appPort,
 		isDaemon:    isDaemon,
 		isResilient: isResilient,
 		ack:         acks[serviceName],
 	}
 }
 
+func (hs *k8sHealthcheckService) populateSystemCodeForService(serviceName string, servicePort int32) {
+	go func(serviceName string, servicePort int32, services *servicesMap, guard chan struct{}) {
+		guard <- struct{}{}
+		defer func(g chan struct{}) {
+			<-g
+		}(guard)
+		log.Infof("Fetching system code for service %s", serviceName)
+
+		req, err := http.NewRequest(http.MethodGet, fmt.Sprintf("http://%s:%d/__health", serviceName, servicePort), nil)
+		if err != nil {
+			log.WithError(err).Errorf("Failed to prepare system code request for service %s", serviceName)
+			return
+		}
+		resp, err := hs.httpClient.Do(req)
+		if err != nil {
+			log.WithError(err).Errorf("Failed to execute system code request for service %s", serviceName)
+			return
+		}
+		if resp.StatusCode != http.StatusOK {
+			log.WithError(err).Errorf("System code request returned non-200 code [%d] for service %s", resp.StatusCode, serviceName)
+			return
+		}
+		defer func() {
+			err := resp.Body.Close()
+			if err != nil {
+				log.WithError(err).Errorf("Failed to close close response body reader for service %s.", serviceName)
+			}
+		}()
+		response := &healthcheckEndpointResponse{}
+		err = json.NewDecoder(resp.Body).Decode(response)
+		if err != nil {
+			log.WithError(err).Errorf("Failed to decode response body for service %s", serviceName)
+			return
+		}
+		services.Lock()
+		defer services.Unlock()
+		if service, ok := services.m[serviceName]; ok {
+			service.sysCode = response.Code
+			services.m[serviceName] = service
+			log.Infof("Fetched code [%s] for service %s", response.Code, serviceName)
+		}
```

Comment:
**potential-issue:** If this code runs before `populateServices` returns the `service` struct which is saved into the `services` map, we won't enter this if. 

Maybe this is the reason why some of the services don't have `systemCode` in heimdal like `list-notifications-push`.